app/client/rest/base.ts

```diff
@@ -303,7 +308,7 @@ export default class ClientBase {
         }
 
         if (response.ok) {
-            return returnDataOnly ? response.data : response;
+            return returnDataOnly ? (response.data || {}) : response;
```

Comment:
can you elaborate here? I understand that previously it was retuning `undefined` why isn't  that desired anymore ?

Reply:
It simplifies typing. But yes, it is a change in behaviour. 0/5 on reverting this change, but it would need lots of undefined checks somewhere else.

Reply:
I rather revert this unless we check and make changes as needed